plugins/MauticCrmBundle/Command/PushDataToPipedriveCommand.php

```diff
@@ -77,18 +80,75 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $this->io->text('Pushed '.$pushed);
         }
 
-        $leads = $em->getRepository(Lead::class)->findAll();
         $this->io->title('Pushing Leads');
+
         $leadExport = $this->getContainer()->get('mautic_integration.pipedrive.export.lead');
         $leadExport->setIntegration($integrationObject);
-        $pushed = 0;
-        foreach ($leads as $lead) {
-            if ($leadExport->create($lead)) {
-                ++$pushed;
+
+        $pushed   = 0;
+        $start    = 0;
+        $limit    = 50;
+        $progress = ProgressBarHelper::init($output, $limit);
+        while (true) {
+            $leads = $this->getLeads($em, $start, $limit);
+
+            if (!$leads) {
+                break;
+            }
+
+            foreach ($leads as $lead) {
+                if ($leadExport->create($lead)) {
+                    ++$pushed;
+                    if ($pushed % $limit == 0) {
+                        $progress->setProgress($pushed);
+                    }
+                }
             }
+            $start = $start + $limit;
+            $em->clear();
         }
-        $this->io->text('Pushed '.$pushed);
 
+        $progress->finish();
+
+        $output->writeln('');
+        $this->io->text('Pushed total '.$pushed);
         $this->io->success('Execution time: '.number_format(microtime(true) - $_SERVER['REQUEST_TIME_FLOAT'], 3));
     }
+
+    /**
+     * @param EntityManager $em
+     * @param               $start
+     * @param               $limit
+     *
+     * @return array
+     */
+    private function getLeads(EntityManager $em, $start, $limit)
+    {
+        return $em->getRepository(Lead::class)->getEntities(
+            [
+                'filter' => [
+                    'force' => [
+                        [
+                            'column' => 'l.email',
+                            'expr'   => 'neq',
+                            'value'  => '',
```

Comment:
Will this work for `NULL` values?

Reply:
It shoudl. I tested it and works. 
Anyway I replaced it with isNotNull expr. 
It's make sense more.